src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,395 @@
+        }
+
+        private bool ArgsIsArrayOfReferenceTypes(
+            SemanticModel semanticModel,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            var argsArgumentType = TryGetArgsArgumentType(semanticModel, arguments, parameters);
+            if (argsArgumentType == null)
+            {
+                return false;
+            }
+
+            if (argsArgumentType.TypeKind != TypeKind.Array)
+            {
+                return false;
+            }
+
+            var argsArray = (IArrayTypeSymbol)argsArgumentType;
+            return argsArray.ElementType.IsReferenceType;
+        }
+
+        private ITypeSymbol TryGetArgsArgumentType(
+            SemanticModel semanticModel,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            var argsArgument = TryGetArgument(semanticModel, NameOfArgsParameter, arguments, parameters);
+            return (argsArgument == null? null : GetArgumentExpressionType(semanticModel, argsArgument));
+        }
+
+        protected SyntaxNode TryGetArgument(
+            SemanticModel semanticModel,
+            string searchArgumentName,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+
+            // First, look for a named argument that matches
+            var matchingNamedArgument = GetMatchingNamedArgument(arguments, searchArgumentName);
+            if (matchingNamedArgument != null)
+            {
+                return matchingNamedArgument;
+            }
+
+            // If no named argument exists, look for the named parameter
+            // and return the corresponding argument
+            var parameterWithMatchingName = GetParameterWithMatchingName(parameters, searchArgumentName);
+            if (parameterWithMatchingName == null)
+            {
+                return null;
+            }
+
+            // For the case string.Format("Test string"), there is only one argument
+            // but the compiler created an empty parameter array to bind to an overload
+            if (parameterWithMatchingName.Ordinal >= arguments.Count)
+            {
+                return null;
+            }
+
+            // Multiple arguments could have been converted to a single params array, 
+            // so there wouldn't be a corresponding argument
+            if (parameterWithMatchingName.IsParams && parameters.Length != arguments.Count)
+            {
+                return null;
+            }
+
+            return arguments[parameterWithMatchingName.Ordinal];
+        }
+
+        private IParameterSymbol GetParameterWithMatchingName(ImmutableArray<IParameterSymbol> parameters, string searchArgumentName)
+        {
+            foreach (var p in parameters)
+            {
+                if (p.Name.Equals(searchArgumentName))
```

Comment:
(bonus, will work if name is null)  (might be an issue for crazy parameters like __arglist).